sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -142,7 +142,7 @@ case class CreateTableAsSelect[T](
     tableName: String,
     child: LogicalPlan,
     allowExisting: Boolean,
-    desc: Option[T] = None) extends UnaryNode {
+    desc: T) extends UnaryNode {
```

Comment:
The `CreateTableAsSelect` is designed as a common logical plan node, that's why I made the `desc` as `T`, and also the optional parameter. Otherwise, every SQL dialect will implements it's own `CTAS` node(logical plan).
Or is the `CreateTableUsingAsSelect` a more generic interface for the same purpose?